core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -613,7 +613,16 @@ class Partition(val topicPartition: TopicPartition,
   // Only ReplicaAlterDirThread will call this method and ReplicaAlterDirThread should remove the partition
   // from its partitionStates if this method returns true
   def maybeReplaceCurrentWithFutureReplica(): Boolean = {
-    // lock to prevent the log append by followers while checking if the log dir could be replaced with future log.
+    maybeFutureReplicaCaughtUp((futurePartitionLog: UnifiedLog) => {
+      logManager.replaceCurrentWithFutureLog(topicPartition)
+      futurePartitionLog.setLogOffsetsListener(logOffsetsListener)
+      log = futureLog
+      removeFutureLocalReplica(false)
+    })
+  }
+
+  def maybeFutureReplicaCaughtUp(callback: UnifiedLog => Unit): Boolean = {
```

Comment:
"maybeFutureReplicaCaughtUp" doesn't suggest anything to me. Maybe it did? So what?

how about a name for this function like "runCallbackIfFutureReplicaCaughtUp" or something.

Reply:
rename it.